### PR TITLE
feat: identity and claims in users ORPC endpoints

### PR DESCRIPTION
**Pull request by SilVdB**
# Summary

This PR implements comprehensive identity and claims management for user ORPC routes, adding permission-based access control and claim filtering based on user roles and trusted issuer status.

 # What Changed

 ## 🔐 Identity Permissions Middleware

  - New: identity-permissions.middleware.ts - Core middleware for computing user
  permissions based on:
  - Identity manager status (full access)
  - Trusted claim issuer status (filtered access)
  - Regular user status (no access)
  - New: Comprehensive test suite with 295+ lines of test coverage

  ## 📋 Enhanced User Routes

  - Enhanced: user.list.ts - Added identity permissions filtering and claims data
  - Enhanced: user.read.ts - Added identity permissions and claim filtering
  - Enhanced: user.me.ts - Added identity fields to user profile response
  - Updated: Schema definitions to include identity and claims fields

  ## 🧪 Comprehensive Testing

  - New: user.list.spec.ts - 579 lines of integration tests covering:
    - Permission-based user listing
    - Claim filtering by user role
    - Identity manager vs trusted issuer access patterns
  - Enhanced: user.read.spec.ts - Extended integration tests for identity permissions
  - Enhanced: Test fixtures with identity and claims mock data

  # Key Features

  ## Permission System

  - Identity Managers: Full access to all users and claims
  - KYC Trusted Issuers: Can see all users but only KYC claims
  - AML Trusted Issuers: Can see all users but only AML claims
  - Regular Users: No access to user listing/details

 ## Claim Filtering

  - Claims are filtered based on user permissions and trusted topics
  - Identity managers see all claims regardless of topic
  - Trusted issuers only see claims for their authorized topics
  - Maintains claim order and handles edge cases (empty arrays, non-existent topics)

 # Testing Approach

  - Unit Tests: Comprehensive middleware testing with all permission combinations
  - Integration Tests: Full ORPC route testing with realistic scenarios
  - Mock Data: Enhanced test fixtures with identity and claims data
  - Edge Cases: Handles undefined access control, empty claims, and invalid states

 # Breaking Changes

  None - all changes are additive and backwards compatible.

## Summary by Sourcery

Add comprehensive identity and claims management to user ORPC endpoints by integrating on-chain data, computing role-based access controls, and filtering visible claims.

New Features:
- Introduce trustedIssuerMiddleware to fetch and inject authorized claim topics into request context
- Implement identityPermissionsMiddleware to compute and enforce per-user read/write permissions for user data and claims
- Extend user.list, user.read, and user.me endpoints to fetch on-chain identity and claims from TheGraph and include identity, claims, and isRegistered in responses
- Create a dedicated UserSearchResultSchema and optimize user.search to return minimal fields (name, wallet, role) for UI components

Enhancements:
- Replace static user dropdowns in AddTrustedIssuerSheet, ViewUserRoles, and GrantRoleForm with searchable input fields backed by the user.search API
- Refactor useSearchAddresses hook to consistently use user.search with minimum character thresholds

Tests:
- Add integration tests for user.list and user.read covering identity registration status, claim filtering, and permission scenarios
- Enhance user.search specs to validate minimal schema, permissions enforcement, and search result limits
- Introduce registerUserIdentity helper in test fixtures to programmatically set up on-chain identity

### CODE REVIEW

File: kit/dapp/src/orpc/routes/user/routes/user.list.ts [hunk 1]
```diff
@@ -78,22 +137,75 @@ export const list = authRouter.user.list
...
+    // - KYC/AML issuers to see only relevant claims for their workflows
+    // - Clean, role-appropriate user interfaces without information overload
+    let accountsData: z.infer<typeof AccountsResponseSchema> = { accounts: [] };
+    if (walletAddresses.length > 0) {
+      accountsData = await context.theGraphClient.query(READ_ACCOUNTS_QUERY, {
+        input: { walletAddresses },
+        output: AccountsResponseSchema,
+      });
+    }
+
```
[the remark this answers is not in the document]
Comment (SilVdB) [reply kit/dapp/src/orpc/routes/user/routes/user.list.ts:165]: This is handled by middleware

File: kit/dapp/src/orpc/routes/user/routes/user.list.ts [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
 import { authRouter } from "@/orpc/procedures/auth.router";
 import type { User } from "@/orpc/routes/user/routes/user.me.schema";
 import { getUserRole } from "@atk/zod/user-roles";
 import { type AnyColumn, asc, desc, eq } from "drizzle-orm";
+import { z } from "zod";
+
+// GraphQL query to fetch multiple accounts by wallet addresses
+const READ_ACCOUNTS_QUERY = theGraphGraphql(`
+  query ReadAccountsQuery($walletAddresses: [Bytes!]!) {
+    accounts(where: { id_in: $walletAddresses }) {
```
[the remark this answers is not in the document]
Comment (SilVdB) [reply kit/dapp/src/orpc/routes/user/routes/user.list.ts:20]: There will potentially be a lot of users, so fetching all of them at once might cause problem. The main purpose of the query is to feed the user management table, so pagination seems like a good solution
